src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -154,6 +155,12 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 							   "relation does not exist")));
 	}
 
+	/* we need to lock partitions too */
```

Comment:
We don't support partitioned tables with `master_create_distributed_table()`, so do we really need this lock(s)?

```SQL
SELECT master_create_distributed_table('collections_list', 'key', 'hash');
ERROR:  distributing partitioned tables in only supported with create_distributed_table UDF                                                                                                                                 
```

Reply:
Removed this to have a closer in next release.